### PR TITLE
Support for box without spaces

### PR DESCRIPTION
**Pull request by tvanyo**
I've updated the RegEx for todo, todoToken, and todoBox to support an empty box (i.e. []) in addition to a box with a space (i.e. [ ]). I've also needed to update the RegEx for Project, Archive and Comment so that any task with the new delimiter will be highlighted as expected in the editor.

Went ahead and updated the README.md to make it clear there is a difference between [] and [ ].

### CODE REVIEW

File: src/consts.ts [hunk 1]
```diff
@@ -30,14 +30,16 @@ const Consts = {
...
+    // todo, todoToken, and todoBox expression changed to allow for no space, e.g. []
+    todo: /^[^\S\n]*((?!--|––|——)(?:[-❍❑■⬜□☐▪▫–—≡→›✘xX✔✓☑+]|\[[ xX+-]?\])[^\n]*)/,
+    todoToken: /^[^\S\n]*(?!--|––|——)([-❍❑■⬜□☐▪▫–—≡→›✘xX✔✓☑+]|\[[ xX+-]?\])/,
+    todoBox: /^[^\S\n]*((?!--|––|——)(?:[-❍❑■⬜□☐▪▫–—≡→›]|\[ {0,1}\])[^\n]*)/,
     todoCancel: /^[^\S\n]*((?!--|––|——)(?:[✘xX]|\[-\])[^\n]*)/,
     todoDone: /^[^\S\n]*((?:[✔✓☑+]|\[[xX+]\])[^\n]*)/,
-    project: /^(?![^\S\n]*(?!--|––|——)(?:[-❍❑■⬜□☐▪▫–—≡→›✘xX✔✓☑+]|\[[ xX+-]\])[^\n]*)[^\S\n]*(.+:[^\S\n]*)(?:(?=@(?!.+ +[^@]))|$)/,
-    archive: /^(?![^\S\n]*(?!--|––|——)(?:[-❍❑■⬜□☐▪▫–—≡→›✘xX✔✓☑+]|\[[ xX+-]\])[^\n]*)([^\S\n]*Archive:.*$)/,
-    comment: /^(?![^\S\n]*(?!--|––|——)(?:[-❍❑■⬜□☐▪▫–—≡→›✘xX✔✓☑+]|\[[ xX+-]\])[^\n]*)(?![^\S\n]*.+:[^\S\n]*(?:(?=@(?!.+ +[^@]))|$))[^\S\n]*([^\n]+)/,
+    // project, archive, and comment expression changed to not recognize [], "[] Item" get highlighted as a task
```
Comment (fabiospampinato) [inline src/consts.ts]: can you remove this comment?

File: src/consts.ts [hunk 2]
```diff
@@ -30,14 +30,16 @@ const Consts = {
   },
   regexes: {
     impossible: /(?=a)b/,
-    todo: /^[^\S\n]*((?!--|––|——)(?:[-❍❑■⬜□☐▪▫–—≡→›✘xX✔✓☑+]|\[[ xX+-]\])[^\n]*)/,
-    todoToken: /^[^\S\n]*(?!--|––|——)([-❍❑■⬜□☐▪▫–—≡→›✘xX✔✓☑+]|\[[ xX+-]\])/,
-    todoBox: /^[^\S\n]*((?!--|––|——)(?:[-❍❑■⬜□☐▪▫–—≡→›]|\[ \])[^\n]*)/,
+    // todo, todoToken, and todoBox expression changed to allow for no space, e.g. []
```
Comment (fabiospampinato) [inline src/consts.ts]: can you remove this comment?

File: src/consts.ts [hunk 3]
```diff
@@ -30,14 +30,16 @@ const Consts = {
...
+    todo: /^[^\S\n]*((?!--|––|——)(?:[-❍❑■⬜□☐▪▫–—≡→›✘xX✔✓☑+]|\[[ xX+-]?\])[^\n]*)/,
+    todoToken: /^[^\S\n]*(?!--|––|——)([-❍❑■⬜□☐▪▫–—≡→›✘xX✔✓☑+]|\[[ xX+-]?\])/,
+    todoBox: /^[^\S\n]*((?!--|––|——)(?:[-❍❑■⬜□☐▪▫–—≡→›]|\[ {0,1}\])[^\n]*)/,
```
Comment (fabiospampinato) [inline src/consts.ts]: can you do ` ?` instead of ` {0,1}`? For consistency.

File: README.md [hunk 4]
```diff
@@ -11,7 +11,7 @@ Manage todo lists with ease. Powerful, easy to use and customizable.
 - **Easy to use**: you're just a few shortcuts away from becoming a master
 - **Portable**: being a plain text format you can read and edit it using any editor
 - **Custom symbols**: you can replace the default symbols with any of the supported ones
-  - **Box**: - ❍ ❑ ■ ⬜ □ ☐ ▪ ▫ – — ≡ → › [ ]
+  - **Box**: - ❍ ❑ ■ ⬜ □ ☐ ▪ ▫ – — ≡ → › [ ] []
```
Comment (fabiospampinato) [inline README.md]: Can you rearrange the symbols from `[ ] []` to `[] [ ]`?

File: src/consts.ts (hunk 1 above)
Comment (tvanyo) [reply src/consts.ts]: Done

File: src/consts.ts (hunk 2 above)
Comment (tvanyo) [reply src/consts.ts]: Done

File: src/consts.ts (hunk 3 above)
Comment (tvanyo) [reply src/consts.ts]: Done. These RegEx gave me a lot more trouble than they should have. It seemed like I tried what I ultimately came up with and couldn't get it to work. Now it does and is consistent.

File: README.md (hunk 4 above)
Comment (tvanyo) [reply README.md]: Done

**Comment by fabiospampinato** [comment]
Thanks for the PR, I've just integrated it in v1.10.4.

There was a problem when repeatedly toggling the box for todos like `[] foo`.

**Pull request merged**